src/components/ConfirmDeleteModal.tsx

```diff
@@ -0,0 +1,74 @@
+'use client';
+
+import { useState } from 'react';
+
+import Button from '@/components/Button';
+import {
+  Modal,
+  ModalClose,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalTitle,
+} from '@/components/Modal';
+
+interface ConfirmDeleteModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  onConfirm: () => Promise<void>;
+  message?: string;
+  confirmText?: string;
+  cancelText?: string;
+}
+
+export default function ConfirmDeleteModal({
+  isOpen,
+  onClose,
+  onConfirm,
+  message = '정말 삭제하시겠습니까?',
+  confirmText = '삭제하기',
+  cancelText = '취소',
+}: ConfirmDeleteModalProps) {
+  const [loading, setLoading] = useState(false);
+
+  const handleConfirm = async () => {
+    setLoading(true);
+    await onConfirm(); // 외부에서 넘긴 삭제 핸들러 실행
+    setLoading(false);
+  };
```

Comment:
```suggestion
  const handleConfirm = async () => {
    setLoading(true);
    try {
    await onConfirm(); // 외부에서 넘긴 삭제 핸들러 실행
    } catch (err) {
    console.log(err) // 토스트로 변경
    } finally {
    setLoading(false);
  }
  };
```

여기 async가 들어가서 비동기 처리라서 이렇게 처리해도 좋겠네요!